### PR TITLE
Fixed bugs and refactor in new EXWM layer 

### PR DESCRIPTION
**Pull request by moritzschaefer**
The PR for SpacemacsOS/EXWM leads to two errors for me (see last comment in https://github.com/syl20bnr/spacemacs/pull/14714)

This PR fixes them. @be11ng would be happy to hear your thoughts about it, since I am really not experienced in lisp etc.

### CODE REVIEW

**Comment by lebensterben** [comment]
I will fix this later today

File: layers/+window-management/exwm/funcs.el [hunk 1]
```diff
@@ -118,7 +118,7 @@ The started process is also added to `exwm-autostart-process-list'."
            for autostart-dir = (concat dir "/autostart")
            for file in (when (file-exists-p autostart-dir)
                          (directory-files autostart-dir t ".+\\.desktop$"))
-           (collect (xdg-desktop-read-file file))))
```
Comment (lebensterben) [inline layers/+window-management/exwm/funcs.el:121]: Change the function body to
```elisp
(cl-loop with autostart-dirs    =  nil
         with autostart-entries =  nil
         for  xdg-dir           in (cons (xdg-config-home) (xdg-config-dirs))
         for  autostart-dir     =  (expand-file-name (concat xdg-dir "/autostart"))
         when (and (file-exists-p autostart-dir)
                   (not (member autostart-dir autostart-dirs)))
           collect autostart-dir into autostart-dirs
           and nconc (mapcar #'xdg-desktop-read-file
                             (directory-files autostart-dir t ".+\\.desktop$")) into autostart-entries
         finally do (print autostart-entries))
```

**Review by lebensterben — CHANGES_REQUESTED** [review]

File: layers/+window-management/exwm/funcs.el (hunk 1 above)
Comment (lebensterben) [reply layers/+window-management/exwm/funcs.el:121]: A few changes from the original one:
- replaced `file-exists-p` with `file-directory-p`, which return t only when it's an existing directory
- duplicated autostart directories are filtered out

**Comment by moritzschaefer** [comment]
@lebensterben let me know if I should do anything.. :)

**Comment by lebensterben** [comment]
@moritzschaefer 
Can you see the comments I left?

**Comment by lebensterben** [comment]
@moritzschaefer 

Can you verify whether it's working?

**Comment by moritzschaefer** [comment]
Hey @lebensterben. Sorry, I really just wanted to fix these syntax errors and at the moment don't have time for testing and reviewing all the changes. So I'd suggest to just merge it.

**Comment by lebensterben** [comment]
@moritzschaefer 
Don't worry. I will take my time until it's verified working.

**Comment by moritzschaefer** [comment]
I quickly pulled the branch and restarted my system. There are neither errors/warnings nor is there any abnormal behavior. Thanks for the effort!

**Comment by lebensterben** [comment]
@moritzschaefer 
merged. thanks for your contributiuon

**Pull request merged**